src/detect-engine-hsbd.c

```diff
@@ -270,12 +299,20 @@ int DetectEngineInspectHttpServerBody(ThreadVars *tv,
     HtpState *htp_state = (HtpState *)alstate;
     uint32_t buffer_len = 0;
     uint32_t stream_start_offset = 0;
-    const uint8_t *buffer = DetectEngineHSBDGetBufferForTX(tx, tx_id,
-                                                     de_ctx, det_ctx,
-                                                     f, htp_state,
-                                                     flags,
-                                                     &buffer_len,
-                                                     &stream_start_offset);
+    const uint8_t *buffer = NULL;
+
+    if (det_ctx->decompressed_hsbd_buffer) {
+        buffer = det_ctx->decompressed_hsbd_buffer;
```

Comment:
Also, ideally there is NO memory allocation for the decompression most of the time. In the regular server body case memory allocations are only done to grow the buffer. In other cases the memory is simply reused.